### PR TITLE
HYDRA-2663: Ignore test sources in `changedProjectsInCompile`

### PR DESCRIPTION
**Pull request by notxcain**
There's no point in taking into account test sources when running `changedProjectsInCompile`.

### CODE REVIEW

File: src/main/scala/com/elarib/PartialSbtPlugin.scala [hunk 1]
```diff
@@ -144,13 +149,19 @@ object PartialSbtPlugin extends AutoPlugin {
 
         val diffsFiles: Seq[sbt.File] = changeGetter.changes.filterNot(f => isFileExcluded(baseDir)(f, excludedFiles))
 
+        def isExcludedTestFile(file: sbt.File, resolvedProject: ResolvedProject): Boolean =
+          excludeTestFiles && file.getAbsolutePath.startsWith((resolvedProject.base / "src" / "test").getAbsolutePath)
```
Comment (danicheg) [inline src/main/scala/com/elarib/PartialSbtPlugin.scala:153]: Let's retrieve all the test directories from SBT (there could be more than one test directory, for example, java sources) — `Test / sourceDirectory`. WDYT?

Comment (notxcain) [reply src/main/scala/com/elarib/PartialSbtPlugin.scala:153]: I was thinking about it, but there's an issue.
In order to access `Test / sourceDirectories` of a certain `ResolvedProject` we need an instance of `Settings[Scope]`, which is provided by an instance of `BuildStructure`, which we can get by evaluating the `buildStructure` **task**. 
However `changedProjects` is a command and it's defined by appending an instance of `Command` to the `commands` **setting**. 
The problem is that a setting could not depend on a task and rewriting `changedProjects` to a task is way outside of the scope of this PR.

BTW `src/test` covers all **default** test directories including `java` and `scala-*`. Which is our case and let it be a limitation of this plugin :)

File: src/sbt-test/test-projects/multi-module-project/changes/6.log [hunk 2]
```diff
@@ -1 +1,2 @@
-src/libs/testKit/build.sbt
\ No newline at end of file
+src/libs/testKit/build.sbt
+src/libs/lib-1/src/test/scala/Test.scala
```
Comment (fido-node) [inline src/sbt-test/test-projects/multi-module-project/changes/6.log:2]: Just add editorconfig from atlas, please.

**Review by fido-node — APPROVED** [review]

File: src/sbt-test/test-projects/multi-module-project/changes/6.log (hunk 2 above)
Comment (notxcain) [reply src/sbt-test/test-projects/multi-module-project/changes/6.log]: It was there but for some reason was disabled for this project in IDEA.

**Pull request merged**